src/components/Counter/Counter.jsx

```diff
@@ -1,9 +1,33 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState, useReducer } from 'react'
 
 const pinkRGB = `rgb(236, 72, 153)`
+const initialCount = 0
+const ACTIONS = {
+  INCREMENT: 'increment',
+  DECREMENT: 'decrement',
+  RESET: 'reset'
+}
+
+function countReducer(count, action) {
+  switch (action.type) {
+    case ACTIONS.INCREMENT: {
+      return count + 1
+    }
+    case ACTIONS.DECREMENT: {
+      return count - 1
+    }
+    case ACTIONS.RESET: {
+      return initialCount
+    }
+    default: {
+      return initialCount
+    }
+  }
+}
+
 
 export default function Counter() {
-  const [count, setCount] = useState(0)
+  const [count, dispatch] = useReducer(countReducer, initialCount)
   const [currentColor, setCurrentColor] = useState(pinkRGB)
```

Comment:
You can also pull the color into your reducer if you want to handle both states in the same place